### PR TITLE
service: remove exported logger from base implemenation

### PR DESCRIPTION
**Pull request by tychoish**
This is really a kinda mino nitpick, but is an (easy) part of
minimizing the overall surface area of the *service.BaseService
implementation

### CODE REVIEW

File: abci/client/socket_client.go [hunk 1]
```diff
@@ -162,7 +163,7 @@ func (cli *socketClient) recvResponseRoutine(ctx context.Context, conn io.Reader
 			return
 		}
 
-		// cli.Logger.Debug("Received response", "responseType", reflect.TypeOf(res), "response", res)
+		// cli.logger.Debug("Received response", "responseType", reflect.TypeOf(res), "response", res)
```
Comment (creachadair) [inline abci/client/socket_client.go:166]: I suppose we could probably just ditch this.

File: internal/libs/autofile/group.go [hunk 2]
```diff
@@ -88,7 +90,12 @@ func OpenGroup(headPath string, groupOptions ...func(*Group)) (*Group, error) {
 		return nil, err
 	}
 
+	if logger == nil {
```
Comment (creachadair) [inline internal/libs/autofile/group.go:93]: It looks as if the only place we use this function outside the package is in internal/consensus/wal.go, so I think we could just mandate it is not nil and let it panic if someone tries to cheat (as with contexts)

File: libs/pubsub/example_test.go [hunk 3]
```diff
@@ -7,14 +7,17 @@ import (
...
 	"github.com/tendermint/tendermint/libs/pubsub/query"
 )
 
 func TestExample(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-	s := newTestServer(ctx, t)
+	logger := log.TestingLogger()
+
+	s := newTestServer(ctx, logger, t)
```
Comment (creachadair) [inline libs/pubsub/example_test.go:20]: optional nit
```suggestion

	s := newTestServer(ctx, log.TestingLogger(), t)
```

File: libs/pubsub/pubsub.go [hunk 4]
```diff
@@ -133,9 +135,12 @@ type Option func(*Server)
 // NewServer returns a new server. See the commentary on the Option functions
 // for a detailed description of how to configure buffering. If no options are
 // provided, the resulting server's queue is unbuffered.
-func NewServer(options ...Option) *Server {
-	s := new(Server)
-	s.BaseService = *service.NewBaseService(nil, "PubSub", s)
+func NewServer(logger log.Logger, options ...Option) *Server {
+	s := &Server{
+		logger: logger,
+	}
```
Comment (creachadair) [inline libs/pubsub/pubsub.go:141]:
```suggestion
	s := &Server{logger: logger}
```

File: libs/pubsub/pubsub.go [hunk 5]
```diff
@@ -167,8 +172,12 @@ func (s *Server) BufferCapacity() int { return cap(s.queue) }
 // If len(capacities) > 0, its first value is used as the queue capacity.
 //
 // Deprecated: Use SubscribeWithArgs. This method will be removed in v0.36.
-func (s *Server) Subscribe(ctx context.Context,
-	clientID string, query Query, capacities ...int) (*Subscription, error) {
+func (s *Server) Subscribe(
```
Comment (creachadair) [inline libs/pubsub/pubsub.go:175]: I humbly request we not do this where it isn't already needed 😢 . We do it all over, but I find this style unreadable and it isn't forced on us here.

File: libs/pubsub/pubsub_test.go [hunk 6]
```diff
@@ -402,12 +415,10 @@ func TestBufferCapacity(t *testing.T) {
 	require.ErrorIs(t, s.Publish(ctx, "Ironclad"), context.DeadlineExceeded)
 }
 
-func newTestServer(ctx context.Context, t testing.TB) *pubsub.Server {
+func newTestServer(ctx context.Context, logger log.Logger, t testing.TB) *pubsub.Server {
```
Comment (creachadair) [inline libs/pubsub/pubsub_test.go:418]: Nit: I think ctx and t should precede other parameters.
```suggestion
func newTestServer(ctx context.Context, t testing.TB, logger log.Logger) *pubsub.Server {
```

**Review by creachadair — APPROVED** [review]

File: libs/pubsub/pubsub.go (hunk 5 above)
Comment (tychoish) [reply libs/pubsub/pubsub.go:173]: I find the lack of indentation between the arguments and the return values more unreadable than this. I'm happy to have one long line, but what was here before was definitely less readable (for me)

Comment (creachadair) [reply libs/pubsub/pubsub.go:173]: 👍🏼 No objection to a long line, IMO.

**Pull request merged**